spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -640,6 +646,12 @@ structure TestProvider with [LanguageProvider] {
 
 }
 
+structure CancelRequestParams {
+    /// The request id to cancel.
+    @required
+    id: RequestId
```

Comment:
RequestId can be either a string or a number https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#cancelRequest